Plugins/Hashing/include/Acts/Plugins/Hashing/HashingAlgorithm.hpp

```diff
@@ -0,0 +1,45 @@
+// This file is part of the Acts project.
+//
+// Copyright (C) 2024 CERN for the benefit of the Acts project
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+#pragma once
+
+#include "Acts/Plugins/Hashing/AnnoyForwardDeclarations.hpp"
+#include "Acts/Plugins/Hashing/HashingAlgorithmConfig.hpp"
+
+namespace Acts {
+
+template <typename external_spacepoint_t, typename SpacePointContainer>
+class HashingAlgorithm {
+ public:
+  explicit HashingAlgorithm(const HashingAlgorithmConfig& cfg);
+
+  /**
+   * @brief Destroy the object.
+   */
+  ~HashingAlgorithm() = default;
+
+  HashingAlgorithm() = default;
+  HashingAlgorithm(const HashingAlgorithm<external_spacepoint_t,
+                                          SpacePointContainer>&) = delete;
+  HashingAlgorithm<external_spacepoint_t, SpacePointContainer>& operator=(
+      const HashingAlgorithm<external_spacepoint_t, SpacePointContainer>&) =
+      default;
```

Comment:
why do we disallow copy construction but allow copy assignment?

I think this could be dropped and be dealt with by default implicit defaults

Reply:
Done, I dropped them.